addons/purchase/views/purchase_views.xml

```diff
@@ -814,21 +822,54 @@
         <field name="name">purchase.history.tree</field>
         <field name="model">purchase.order.line</field>
         <field name="arch" type="xml">
-            <tree create="false" default_order="order_id asc">
+            <tree create="false" default_order="order_id asc" action="action_open_order" type="object">
+                <field name="currency_id" column_invisible="True"/>
                 <field name="order_id" widget="many2one"/>
                 <field name="date_approve"/>
-                <field name="partner_id"/>
-                <field name="product_uom_qty"/>
-                <field name="price_unit"/>
+                <field name="partner_id" string="Vendor"/>
+                <field name="company_id" optional="show" groups="base.group_multi_company"/>
+                <field name="product_uom_qty" string="Quantity"/>
+                <field name="price_unit" widget="monetary"/>
+                <field name="price_subtotal" widget="monetary" string="Total Untaxed"/>
+                <field name="state" optional="hide"/>
             </tree>
         </field>
     </record>
 
+    <record id="purchase_history_pivot_view" model="ir.ui.view">
+         <field name="name">purchase.history.pivot</field>
+         <field name="model">purchase.order.line</field>
+         <field name="arch" type="xml">
+             <pivot>
+                <field name="price_total" string="Total purchased amount" type="measure"/>
+                <field name="product_uom_qty" string="Total Qty purchased" type="measure"/>
+             </pivot>
+         </field>
+    </record>
+
+     <record id="purchase_history_graph_view" model="ir.ui.view">
+         <field name="name">purchase.history.graph</field>
+         <field name="model">purchase.order.line</field>
+         <field name="arch" type="xml">
+             <graph string="purchase Analysis" type="line" sample="1">
+                <field name="date_order" interval="month"/>
+                <field name="product_uom_qty" type="measure"/>
+             </graph>
+         </field>
+    </record>
+
     <record id="action_purchase_history" model="ir.actions.act_window">
         <field name="res_model">purchase.order.line</field>
-        <field name="view_mode">tree</field>
+        <field name="view_mode">tree,pivot,graph</field>
         <field name="view_id" ref="purchase.purchase_history_tree"/>
-        <field name="context">{}</field>
+        <field name="context">
+            {
+                'search_default_purchase_state': 1,
+                'search_default_order_date_last_year': 1,
+                'search_default_date_order': 1,
+                'search_default_groupby_supplier': 1,
+            }
+        </field>
```

Comment:
```suggestion
        <field name="context">{
            'search_default_purchase_state': 1,
            'search_default_order_date_last_year': 1,
            'search_default_groupby_supplier': 1,
        }</field>
```
look better this way
No need for `search_default_date_order` :thinking: no?

Reply:
Done

Reply:
Yes, there is no need to use  'search_default_date_order' because we are using 'search_default_order_date_last_year'
and we got our desired result :+1: 

Thankyou

Reply:
I think you missed first part :pray:

Reply:
Sorry By mistake i have missed the first part  :cry: